mailet/standard/src/main/java/org/apache/james/transport/mailets/LogMessage.java

```diff
@@ -68,45 +70,45 @@ public String getMailetInfo() {
 
     public void service(Mail mail) {
         log("Logging mail " + mail.getName());
-        if (comment != null) log(comment);
+        comment();
         try {
-            if (headers) log(getMessageHeaders(mail.getMessage()));
-            if (body) {
-                int len = bodyMax > 0 ? bodyMax : mail.getMessage().getSize();
-                StringBuilder text = new StringBuilder(len);
-                InputStream is = mail.getMessage().getRawInputStream();
-                byte[] buf = new byte[1024];
-                int read;
-                while (text.length() < len && (read = is.read(buf)) > -1) {
-                    text.append(new String(buf, 0, Math.min(read, len - text.length())));
-                }
-                log(text.toString());
-            }
-        }
-        catch (MessagingException e) {
+            MimeMessage message = mail.getMessage();
+            headers(message);
+            body(message);
+        } catch (MessagingException e) {
             log("Error logging message.", e);
-        }
-        catch (java.io.IOException e) {
+        } catch (java.io.IOException e) {
             log("Error logging message.", e);
         }
         if (!passThrough) {
             mail.setState(Mail.GHOST);
         }
     }
 
-    /**
-     * Utility method for obtaining a string representation of a
-     * Message's headers
-     * 
-     * @param message
-     */
-    private String getMessageHeaders(MimeMessage message) throws MessagingException {
-        @SuppressWarnings("unchecked")
-        Enumeration<String> heads = message.getAllHeaderLines();
-        StringBuilder headBuffer = new StringBuilder(1024).append("\n");
-        while(heads.hasMoreElements()) {
-            headBuffer.append(heads.nextElement().toString()).append("\n");
+    private void comment() {
+        if (comment != null) {
+            log(comment);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void headers(MimeMessage message) throws MessagingException {
+        if (headers) {
+            log("\n");
+            for (String header : Collections.list((Enumeration<String>) message.getAllHeaderLines())) {
+                log(header + "\n");
+            }
         }
-        return headBuffer.toString();
+    }
+
+    private void body(MimeMessage message) throws MessagingException, IOException {
+        if (body) {
+            InputStream inputStream = message.getRawInputStream();
```

Comment:
Which one? The mime4j one?
It's htrowing an IOException when limit is reached, i don't want that.